### ISSUE TITLE
sound: create final demo example of text to speech library

### ISSUE THREAD

**Issue by parkerbroz** [uchicago-cs/chiventure#1437]
Summary: In this issue, we will be creating an example of what text to speech would look like in the chiventure game. This will be done by taking the text to speech libraries we have found, and creating an example using those data structures, and leaving room to build upon it for future teams. We are thinking this could be done by creating an example game in chiventure and using the text to speech functions to create sound through the computer speakers of the text on the in-game screen. Perhaps in the future, it would be worth generalizing this process into specific functions in order to increase ease of implementation.

assignees: Everyone

**Comment by JaStarbuck** [comment]
From slack, brainstorming ideas for what a demo would look like:
We have gotten the demo from their github to work--if all else fails we could show them that and just have the speech say something related to chiventure. Hopefully we can create a function that given some text will say something so we can make a demo game that uses the function and will text to speech what is on the screen during the game.

**Comment by JaStarbuck** [comment]
from meeting with Elizabeth: priority is to create a demo in sandbox, if that gets done, then create functions

**Comment by JaStarbuck** [comment]
I have approved pull request #1591 which will allow us to integrate text to speech now. #1591 was creating a demo game with dialogue that we can transition into text to speech. This is good news.

**Comment by sangerr** [comment]
The pull request for creating a demo game for dialogue has been approved. In order to add text to speech into this demo, we need to create the structs for the text to speech. We are currently working on creating these structs in issue #1596. The demo without the text to speech is located in src/sound/examples under tts_dialogue_example.c

**Comment by sangerr** [comment]
The tts structs have been implemented, and we are just awaiting approval on the pull request

**Comment by sangerr** [comment]
Just created a pull request with the final demo with text to speech functionality, now just waiting on approval.

**Comment by jwschroeder21** [comment]
Update: we are now getting an error with compilation in our include paths that we are trying to fix. The error has something to do with the e-speak library that we will clone into a different repo.

**Comment by sangerr** [comment]
Closing Statement: In this issue, we created an example of what text to speech would look like in the chiventure game. We monkeypatched it into a sample conversation between an npc and individual, where the dialogue from the npc is read aloud. This is a great proof of concept of the potential for text to speech to alter the game experience. We had some challenges with compiling and building the demo along the way, but were able to resolve those as a team! This is our final issue of the quarter and we wanted to say thank you to Elizabeth for being an awesome TA <3

Note: this issue will be closed once the PR is merged
